src/network/MissionControlProtocol.cpp

```diff
@@ -138,14 +139,33 @@ void MissionControlProtocol::handleSetArmIKEnabled(const json& j) {
 		}
 		DataPoint<navtypes::Vectord<Constants::arm::IK_MOTORS.size()>> armJointPositions =
 			robot::getMotorPositionsRad(Constants::arm::IK_MOTORS);
+
 		// TODO: there should be a better way of handling invalid data than crashing.
 		// It should somehow just not enable IK, but then it needs to communicate back to MC
 		// that IK wasn't enabled?
 		assert(armJointPositions.isValid());
-		Globals::planarArmController.set_setpoint(armJointPositions.getData());
-		Globals::armIKEnabled = true;
-		_arm_ik_repeat_thread =
-			std::thread(&MissionControlProtocol::updateArmIKRepeatTask, this);
+
+		// Reject enabling IK if EE outside of safety radius.
+		// Compute the new EE position to determine if it is within safety factor.
+		// If so, then enable IK by emplacing Globals::planarArmController and setting the
+		// setpoint. Otherwise, is a no-op.
+		double eeRadius =
+			Globals::planarArmKinematics.jointPosToEEPos(armJointPositions.getData()).norm();
+		double maxRadius =
+			Globals::planarArmKinematics.getSegLens().sum() * Constants::arm::SAFETY_FACTOR;
+
+		if (eeRadius <= maxRadius) {
+			if (!Globals::planarArmController.has_value()) {
+				Globals::planarArmController.emplace(navtypes::Vectord<2>(0, 0),
+													 Globals::planarArmKinematics,
+													 Constants::arm::SAFETY_FACTOR);
+			}
```

Comment:
This leaks implementation details a little bit, this should maybe be a factory function in `PlanarArmController`.

So maybe we could do:
```c++
if (!Globals::planarArmController.has_value()) {
    Globals::planarArmController = control::PlanarArmController::makeController(...)
}

if (Globals::planarArmController.has_value()) {
    // init thread, etc
}
```

and `makeController()` would check if the arm is in a valid position, and if it is, instantiate and return a controller, and if it isn't, return an empty optional.